seacrowd/sea_datasets/creole_rc/creole_rc.py

```diff
@@ -0,0 +1,197 @@
+            features = schemas.kb_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        data_paths = {
+            "csv": Path(dl_manager.download_and_extract(_URLS["csv"])),
+            "json": Path(dl_manager.download_and_extract(_URLS["json"])),
+        }
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "csv_filepath": data_paths["csv"],
+                    "json_filepath": data_paths["json"],
+                    "split": "train",
+                },
+            )
+        ]
+
+    # method parameters are unpacked from `gen_kwargs` as given in `_split_generators`
+
+    def _generate_examples(self, csv_filepath: Path, json_filepath: Path, split: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+        # read csv file
+        with open(csv_filepath, "r", encoding="utf-8") as csv_file:
+            csv_reader = csv.reader(csv_file)
+            csv_data = [row for row in csv_reader]
+        csv_data = csv_data[1:]  # remove header
+
+        # read json file
+        with open(json_filepath, "r", encoding="utf-8") as json_file:
+            json_data = json.load(json_file)
+
+        # properties descriptions from https://github.com/hclent/CreoleVal/tree/main/nlu/relation_classification
+        # for properties present in Chavacano subset
+        properties_desc = {"P17": "country", "P30": "continent", "P106": "occupation", "P131": "located in the administrative territorial entity", "P361": "part of", "P1376": " capital of country"}
+
+        num_sample = len(csv_data)
+
+        for i in range(num_sample):
+            if self.config.schema == "source":
+                example = {
+                    "id": str(i),
+                    "sentence": csv_data[i][0],
+                    "ent1": csv_data[i][1],
+                    "ent2": csv_data[i][2],
+                    "ent1_qcode": csv_data[i][3],
+                    "ent2_qcode": csv_data[i][4],
+                    "property": csv_data[i][5],
+                    "property_desc": properties_desc[csv_data[i][5]],
+                    "tokens": json_data[i]["tokens"],
+                    "edgeset_left": json_data[i]["edgeSet"]["left"],
+                    "edgeset_right": json_data[i]["edgeSet"]["right"],
+                    "edgeset_triple": json_data[i]["edgeSet"]["triple"],
+                }
+            elif self.config.schema == f"seacrowd_{self.SEACROWD_SCHEMA_NAME}":
+                offset_entity1 = csv_data[i][0].find(csv_data[i][1])
+                offset_entity2 = csv_data[i][0].find(csv_data[i][2])
```

Comment:
Can we put a sanity check if either of these values isn't found, leading to a -1 value? we can skip those rows if that's the case